### DISCUSSION TITLE
Trying to use includemaxlength and $wgDplSettings

### DISCUSSION THREAD

**Discussion by Lucire** [Universal-Omega/DynamicPageList4#122] [Q&A] [answered]
Hi there, I had previously used the old DynamicPageList with an older version of Mediawiki (I think 1.22). We recently upgraded on a new server and began using DynamicPageList3. 

There are a couple of things that don’t seem to work, and as I’m not technical, most likely I am missing something.

Here is the code inside my Mediawiki installation. I carried over what I thought was relevant from the old installation.

```
<DPL>
namespace= 
notcategory=Nameplates
notcategory=Marques
notcategory=Information
include=*
includemaxlength=[300]
escapelinks=false 
randomcount=1 
mode=userformat 
addpagecounter=true 
listseparators=<h3>,[[%PAGE%]]</h3>,,\n\n
</DPL>
```

The hope is to only show the first 300 characters from each page.

I have tried `includepages=*` and `includemaxlength=300` as well.

The result is, sadly, I get the entire page cited, not the first 300 characters.

Also the pages DynamicPageList3 seems to choose begin with either A or B. 

I remember I had this problem with the original DynamicPageList and looking through the old LocalSettings.php, I believe I added these lines:

```
include("$IP/extensions/DynamicPageList/DynamicPageList.php");
ExtDynamicPageList::$allowUnlimitedResults = true;
```

I looked up the equivalent for DynamicPageList3 and I understand the equivalent to be:

```
wfLoadExtension( 'DynamicPageList3' );
$wgDplSettings['allowUnlimitedCategories'] = true;
$wgDplSettings['allowUnlimitedResults']    = true;

```
Yet I cannot get pages that don’t begin with A and B.

Could you please advise what I am doing wrong?

**Comment by Lucire** [comment]
I browsed this site a bit, and it seems allowUnlimitedResults had been discussed. I have now made this change in ParametersData.php:

```
		/**
		 * Max of results to display, selection is based on random.
		 */
		'count' => [
			'default' => 50000,
			'integer' => true
```

Now trying to sort the other thing … thank you for keeping track of all these discussions and changes!

**Reply by Universal-Omega** [reply]
Adding count = 50000 to DPL tag should work instead of needing to modify ParametersData. As I've recently fixed a lot of issues with some configuration options like allowUnlimitedResults so they should now work.

**Reply by Lucire** [reply]
Thank you, that was one thing I didn’t try!

**Reply by Lucire** [reply]
I re-downloaded DPL3 and reinstalled, but it didn’t seem to work. I could have been putting the count line into the wrong place (added after the `includemaxlength=[300]` line), but changing ParametersData still worked for me.

**Comment by Universal-Omega** [comment, marked as answer]
* https://github.com/Universal-Omega/DynamicPageList3/pull/123 should fix the issue with `includemaxlength` (and unrelated `titlemaxlength`) since that was an issue with the extension as far as I can tell. The fix is untested, though, feel free to test that out and let me know, otherwise it'll be awhile before that fix is deployed.

**Reply by Lucire** [reply]
Excellent, thank you! Let me give it a go and I’ll come back to you.

**Reply by Lucire** [reply]
Hmm, I get these errors. Maybe I am using a different DynamicPageList3? Or should I re-download all the files to make sure I am using the latest?

[93d3f67e29a42b0e6b453b67] /index.php/Main_Page Error from line 12 of /var/www/vhosts/autocade.net/httpdocs/extensions/DynamicPageList3/includes/ParametersData.php: Class 'MediaWiki\Extension\DynamicPageList3\ParametersData' not found

Backtrace:

#0 /usr/share/mediawiki/includes/AutoLoader.php(109): require()
#1 [internal function]: AutoLoader::autoload()
#2 /var/www/vhosts/autocade.net/httpdocs/extensions/DynamicPageList3/includes/Parameters.php(11): spl_autoload_call()
#3 /usr/share/mediawiki/includes/AutoLoader.php(109): require(string)
#4 [internal function]: AutoLoader::autoload()
#5 /var/www/vhosts/autocade.net/httpdocs/extensions/DynamicPageList3/includes/Parse.php(103): spl_autoload_call()
#6 /var/www/vhosts/autocade.net/httpdocs/extensions/DynamicPageList3/includes/DynamicPageListHooks.php(203): DPL\Parse->__construct()
#7 /var/www/vhosts/autocade.net/httpdocs/extensions/DynamicPageList3/includes/DynamicPageListHooks.php(187): DPL\DynamicPageListHooks::executeTag()
#8 /usr/share/mediawiki/includes/parser/Parser.php(3922): DPL\DynamicPageListHooks::dplTag()
#9 /usr/share/mediawiki/includes/parser/PPFrame_Hash.php(340): Parser->extensionSubstitution()
#10 /usr/share/mediawiki/includes/parser/Parser.php(2887): PPFrame_Hash->expand()
#11 /usr/share/mediawiki/includes/parser/Parser.php(1556): Parser->replaceVariables()
#12 /usr/share/mediawiki/includes/parser/Parser.php(651): Parser->internalParse()
#13 /usr/share/mediawiki/includes/content/WikitextContent.php(374): Parser->parse()
#14 /usr/share/mediawiki/includes/content/AbstractContent.php(590): WikitextContent->fillParserOutput()
#15 /usr/share/mediawiki/includes/Revision/RenderedRevision.php(263): AbstractContent->getParserOutput()
#16 /usr/share/mediawiki/includes/Revision/RenderedRevision.php(235): MediaWiki\Revision\RenderedRevision->getSlotParserOutputUncached()
#17 /usr/share/mediawiki/includes/Revision/RevisionRenderer.php(215): MediaWiki\Revision\RenderedRevision->getSlotParserOutput()
#18 /usr/share/mediawiki/includes/Revision/RevisionRenderer.php(152): MediaWiki\Revision\RevisionRenderer->combineSlotOutput()
#19 [internal function]: MediaWiki\Revision\RevisionRenderer->MediaWiki\Revision\{closure}()
#20 /usr/share/mediawiki/includes/Revision/RenderedRevision.php(197): call_user_func()
#21 /usr/share/mediawiki/includes/poolcounter/PoolWorkArticleView.php(216): MediaWiki\Revision\RenderedRevision->getRevisionParserOutput()
#22 /usr/share/mediawiki/includes/poolcounter/PoolCounterWork.php(162): PoolWorkArticleView->doWork()
#23 /usr/share/mediawiki/includes/page/Article.php(810): PoolCounterWork->execute()
#24 /usr/share/mediawiki/includes/actions/ViewAction.php(80): Article->view()
#25 /usr/share/mediawiki/includes/MediaWiki.php(532): ViewAction->show()
#26 /usr/share/mediawiki/includes/MediaWiki.php(313): MediaWiki->performAction()
#27 /usr/share/mediawiki/includes/MediaWiki.php(945): MediaWiki->performRequest()
#28 /usr/share/mediawiki/includes/MediaWiki.php(548): MediaWiki->main()
#29 /usr/share/mediawiki/index.php(53): MediaWiki->run()
#30 /usr/share/mediawiki/index.php(46): wfIndexMain()
#31 {main}

**Reply by Universal-Omega** [reply]
That would be because you aren't using latest master, then separately, that PR contents. This is because there has been a ton of recent changes to DPL3 lately that it would differ if you copies and pasted the entire Parameters.php.

**Reply by Lucire** [reply]
Perfect, thank you! The only change I made to my tags was `includemaxlength=300` and not `includemaxlength=[300]` as I had when I first commented on this thread.

**Reply by Universal-Omega** [reply]
it should be using includemaxlength=[300]... oh wait I think I did something wrong here then. I think I know why though. Glad that worked for you though. I'll deploy that fix and later fix to use with brackets as that is what the documentation states.

**Reply by Universal-Omega** [reply]
I have now released this fix under the new version 3.4.1 of DPL3. See https://github.com/Universal-Omega/DynamicPageList3/releases/tag/3.4.1

**Comment by Universal-Omega** [comment]
* #131 should hopefully fix `allowUnlimitedResults` also. (Without changing ParametersData)

**Reply by Lucire** [reply]
Awesome, thank you!
